src/screens/validator-screen/validator-screen.tsx

```diff
@@ -56,11 +77,39 @@ const navigationOptions = ({ navigation }: any) => ({
 
 export class ValidatorScreenComponent extends React.Component<
     INavigationParams &
+        ReduxProps &
         INavigationProps<INavigationParams> &
-        IThemeProps<ReturnType<typeof stylesProvider>>
+        IThemeProps<ReturnType<typeof stylesProvider>>,
+    State
 > {
+    constructor(props) {
+        super(props);
+        this.state = {
+            total: new BigNumber(0)
+        };
+    }
+
     public static navigationOptions = navigationOptions;
 
+    componentDidMount() {
+        this.getBalance();
+    }
+
+    async getBalance() {
+        const { total } = await getBlockchain(this.props.blockchain)
+            .getClient(this.props.chainId)
+            .getBalance(this.props.account.address);
+        this.setState({ total });
+    }
+
+    handleOnClaimAlert() {
+        const config = getBlockchain(this.props.blockchain).config;
```

Comment:
i think we can move this in can perform action, and just return an error code. Based on the error code we can display the texts in the warning popup